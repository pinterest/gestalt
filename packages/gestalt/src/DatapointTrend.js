// @flow strict
import { type Node } from 'react';
import Text from './Text.js';
import Flex from './Flex.js';
import Icon from './Icon.js';

const TREND_COLOR_MAP = {
  good: 'pine',
  bad: 'red',
  neutral: 'darkGray',
};

type Sentiment = 'good' | 'bad' | 'neutral' | 'auto';

function getValueColor({ sentiment, value }) {
  if (sentiment === 'auto') {
    if (value === 0) {
      return 'darkGray';
    }
    return value > 0 ? 'pine' : 'red';
  }
  return TREND_COLOR_MAP[sentiment];
}

type Props = {|
  iconAccessibilityLabel: string,
  sentiment?: Sentiment,
  value: number,
|};

export default function DatapointTrend({
  iconAccessibilityLabel,
  sentiment = 'auto',
  value,
}: Props): Node {
  const color = getValueColor({ sentiment, value });

  return (
    <Flex gap={1}>
      {value !== 0 && (
        <Icon
          accessibilityLabel={iconAccessibilityLabel}
          color={color}
          icon={value > 0 ? 'sort-ascending' : 'sort-descending'}
          size={16}
        />
      )}

      <Text size="md" color={color} weight="bold">
        {`${Math.abs(value)}%`}
      </Text>
    </Flex>
  );
}
