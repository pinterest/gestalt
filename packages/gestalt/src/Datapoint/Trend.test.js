// @flow strict
import { create } from 'react-test-renderer';
import DatapointTrend from './Trend';

describe('DatapointTrend', () => {
  it('renders', () => {
    const tree = create(
      <DatapointTrend
        iconAccessibilityLabel="Value change icon accessibility label"
        sentiment="good"
        value={30}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders an accessibility label', () => {
    const tree = create(
      <DatapointTrend iconAccessibilityLabel="Value change icon accessibility label" value={30} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
