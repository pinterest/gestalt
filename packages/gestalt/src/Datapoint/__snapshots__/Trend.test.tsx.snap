// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatapointTrend renders 1`] = `
<div
  className="Flex rowGap1 columnGap0 xsDirectionRow"
>
  <div
    className="FlexItem"
  >
    <svg
      aria-hidden={null}
      aria-label="Value change icon accessibility label"
      className="success icon iconBlock"
      height={16}
      role="img"
      viewBox="0 0 24 24"
      width={16}
    >
      <path
        d="test-file-stub"
      />
    </svg>
  </div>
  <div
    className="FlexItem"
  >
    <div
      className="success alignStart breakWord Text fontSize200 fontWeightSemiBold"
    >
      30%
    </div>
  </div>
</div>
`;

exports[`DatapointTrend renders an accessibility label 1`] = `
<div
  className="Flex rowGap1 columnGap0 xsDirectionRow"
>
  <div
    className="FlexItem"
  >
    <svg
      aria-hidden={null}
      aria-label="Value change icon accessibility label"
      className="success icon iconBlock"
      height={16}
      role="img"
      viewBox="0 0 24 24"
      width={16}
    >
      <path
        d="test-file-stub"
      />
    </svg>
  </div>
  <div
    className="FlexItem"
  >
    <div
      className="success alignStart breakWord Text fontSize200 fontWeightSemiBold"
    >
      30%
    </div>
  </div>
</div>
`;
